src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4942,10 +4942,10 @@ public ReadOnlyCollection<string> PayloadNames
             get
             {
                 // For contract based events we create the list lazily.
-                if (m_payloadNames == null)
+                // You can have m_payloadNames be null in the TraceLogging case (EventID < 0) so only
+                // do the lazy init if you know it is contract based (EventID >= 0)
+                if (EventId >= 0 && m_payloadNames == null)
```

Comment:
Your question only matters for the events written with Write<T> and which have no properties (thus no property names).   The question is what happens in that case.     Previously, PayloadNames threw an exception.    It is reasonable to make it either return null or an empty collection.   

There is a Payload property that is the list of values corresponding to the names.  It currently returns null for the Write<T> case, which is why I made this one do it as well.  

However it is also true that BOTH of these properties return an empty list if the event was logged with WriteEvent rather than Write<T>.   Thus you can argue that we should fix it so that Write<T> (For both properties) returns an empty list.   

My original intent was to quickly fix a obvious bug, not to fix a API consistency issue.   

It is not hard to fix.   What do you think?    My inclination is not to spend more time, but if you think it is worth it I will make that modification as well.